[CR]Give acid pools some arpen

Coolthulhu opened the pull request:
(Relatively) Recent buff to material acid resistance made acid deal no damage to even modestly armored player.
In fact, it made spitters more dangerous to their friends than to players.

I changed the acid pool effect to have percentage armor penetration dependent on player's environmental protection on affected parts. This makes regular clothing relatively useless against acid (as it should be), keeps hazmat and rubber protective, but doesn't make acid destroy armor instantly.
Survivor armor is a fair bit too effective here (10 env vs rubber's 12), but I didn't nerf it or try to get around it here.

Gave enemies some acid resistance - for now equal to 1/2 of cutting armor. Most cut-resistant monsters wear kevlar or are made of steel, so giving them some acid resistance approximates what player would have.

Fixed a bug that caused acid fields to be considered safe for the player to walk into.

BevapDin commented on src/player.cpp:
```diff
@@ -12401,48 +12401,54 @@ void get_armor_on(player* p, body_part bp, std::vector<int>& armor_indices) {
-        if( !one_in( num_parts_covered ) ) {
-            return;
-        }
+    if (rng(0,100) > armor.get_coverage()) {
```
Shouldn't this be a `>=`? Coverage of 0 should give no change (currently it has a 1% chance).

And it should probably be rng(0,99) because otherwise (with `>=`) even 100 would not grant complete protection.

Or is this all by design and works as intended?

Coolthulhu replied:
It probably wasn't by design - a lot of old code didn't get reviewed for a loooong time (boost to damage from crits only recently started working, for example).
Will fix

kevingranade replied:
Well crap, I fixed the issue where 100% coverage always had a 1% miss chance, but missed the fact that it had the same error on the other end.  I'm not sure there is anything with 0% coverage though, "trinkets" with no protection and no encumbrance are generally not assigned to a body part at all.  It'd be pretty funny if there was some "lucky charm" that really did have a 1% chance of intercepting some damage though.

Rivet-the-Zombie commented:
I like this a lot.

kevingranade commented on src/field.cpp:
```diff
@@ -1635,32 +1635,39 @@ void map::player_in_field( player &u )
             }
-            if (cur->getFieldDensity() > 0 && !u.has_trait("ACIDPROOF")) {
+            if( cur->getFieldDensity() > 0 && !u.has_trait("ACIDPROOF") ) {
                 u.add_msg_player_or_npc(m_bad, _("The acid burns your legs and feet!"), _("The acid burns <npcname>s legs and feet!"));
```
It looks like there's a chance for the player to take no damage due to environmental resistance, but this message will still print.

Coolthulhu commented:
Changed the message. When the acid doesn't manage to deal any damage to the player (because of acidproof mutation or good boots), the message will simply be a warning.

Changed the armor bypass formula to `if( rng( 1, 100 ) <= coverage ) return;`. That way 0% coverage never protects, 100% coverage always does and there are 100 possible numbers to roll.
